include/Operator/SecondQuantized/USQHamiltonian.hpp

```diff
@@ -0,0 +1,250 @@
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-gqcp.  If not, see <http://www.gnu.org/licenses/>.
+// 
+#pragma once
+
+
+#include "Operator/SecondQuantized/SQHamiltonian.hpp"
+
+
+namespace GQCP {
+
+
+/**
+ *  A class for representing the unrestricted second-quantized electronic Hamiltonian
+ *  It consists of two restricted Hamiltonians for both the beta and alpha component and a mixed two-electron term
+ * 
+ *
+ *  @tparam Scalar      the scalar type of the second-quantized parameters (i.e. the integrals)
+ */
+template <typename Scalar>
+class USQHamiltonian {
+private:
+    SQHamiltonian<Scalar> sq_hamiltonian_alpha;  // alpha hamiltonian
+    SQHamiltonian<Scalar> sq_hamiltonian_beta;  // beta hamiltonian
+
+    std::vector<ScalarSQTwoElectronOperator<Scalar>> two_op_mixed;  // alpha & beta mixed operators represented as g_aabb
+    ScalarSQTwoElectronOperator<Scalar> total_two_op_mixed;  // the total alpha & beta mixed operator represented as g_aabb
+
+public:
+
+    /*
+     *  CONSTRUCTORS
+     */
+    USQHamiltonian() = default;
+
+    /**
+     *  @param sq_hamiltonian_alpha      alpha hamiltonian
+     *  @param sq_hamiltonian_beta       beta hamiltonian
+     *  @param two_op_mixed              alpha & beta mixed operators
+     */
+    USQHamiltonian(const SQHamiltonian<Scalar>& sq_hamiltonian_alpha, const SQHamiltonian<Scalar>& sq_hamiltonian_beta, const std::vector<ScalarSQTwoElectronOperator<Scalar>>& two_op_mixed) :
+        sq_hamiltonian_alpha (sq_hamiltonian_alpha),
+        sq_hamiltonian_beta (sq_hamiltonian_beta),
+        two_op_mixed (two_op_mixed)
+    {
+
+        const auto dim = sq_hamiltonian_alpha.dimension();
+
+        if (sq_hamiltonian_beta.dimension() != dim) {
+            throw std::invalid_argument("USQHamiltonian::USQHamiltonian(const SQHamiltonian<Scalar>& sq_hamiltonian_alpha, const SQHamiltonian<Scalar>& sq_hamiltonian_beta, const ScalarSQTwoElectronOperator<Scalar>& two_op_mixed): The dimensions of the alpha and beta Hamiltonian are incompatible");
+        }
+        
+        for (const auto& two_op : this->two_op_mixed) {
+            if (two_op.dimension() != dim) {
+                throw std::invalid_argument("USQHamiltonian::USQHamiltonian(const SQHamiltonian<Scalar>& sq_hamiltonian_alpha, const SQHamiltonian<Scalar>& sq_hamiltonian_beta, const ScalarSQTwoElectronOperator<Scalar>& two_op_mixed): The dimensions of the mixed two electron operator are incompatible with the Hamiltonian");
+            }
+        }
+        
+        // Calculate the total two-electron operator
+        QCRankFourTensor<Scalar> total_two_op_par (dim);
+        total_two_op_par.setZero();
+        for (const auto& two_op : this->two_op_mixed) {
+            total_two_op_par += two_op.parameters().Eigen();
+        }
+        this->total_two_op_mixed = ScalarSQTwoElectronOperator<Scalar>({total_two_op_par});
```

Comment:
I think two-electron operators can actually be added using `operator+`, see the following implementation that we have: 

https://github.com/GQCG/gqcp/blob/8fe6e43c4a10963545305f429d57255ef55f2aa2/include/Operator/SecondQuantized/SQTwoElectronOperator.hpp#L351-L362